mod/state-transition/pkg/core/state_processor_genesis_test.go

```diff
@@ -298,25 +348,30 @@ func commonChecksValidators(
 
 	idx, err := bs.ValidatorIndexByPubkey(dep.Pubkey)
 	require.NoError(t, err)
-	require.Equal(t, math.U64(dep.Index), idx)
 
 	val, err := bs.ValidatorByIndex(idx)
 	require.NoError(t, err)
 	require.Equal(t, dep.Pubkey, val.Pubkey)
 
 	var (
 		maxBalance = math.Gwei(cs.MaxEffectiveBalance())
-		minBalance = math.Gwei(cs.EffectiveBalanceIncrement())
+		increment  = math.Gwei(cs.EffectiveBalanceIncrement())
+		minBalance = math.Gwei(cs.EjectionBalance())
 	)
 	switch {
 	case dep.Amount >= maxBalance:
 		require.Equal(t, maxBalance, val.EffectiveBalance)
-	case dep.Amount >= minBalance && dep.Amount < maxBalance:
-		require.Equal(t, dep.Amount, val.EffectiveBalance)
-
-		// validator balance must be multiple of EffectiveBalanceIncrement
-		require.Equal(t, math.U64(0), val.EffectiveBalance%minBalance)
-	case dep.Amount < minBalance:
+	case dep.Amount > minBalance && dep.Amount < maxBalance:
+		// Effective balance must be a multiple of increment.
+		// If balance is not, effective balance is rounded down
+		if dep.Amount%increment == 0 {
```

Comment:
improved checks here. Hopefully the effective balance amount invariants are clearer